src/test/java/io/openshift/booster/OpenshiftIT.java

```diff
@@ -18,51 +18,31 @@
 
 package io.openshift.booster;
 
-import java.util.concurrent.TimeUnit;
+import java.net.URL;
 
 import com.jayway.restassured.RestAssured;
-import com.jayway.restassured.response.Response;
-import io.openshift.booster.test.OpenShiftTestAssistant;
-import org.junit.AfterClass;
-import org.junit.BeforeClass;
+import org.arquillian.cube.openshift.impl.enricher.RouteURL;
+import org.jboss.arquillian.junit.Arquillian;
+import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
 
-import static com.jayway.awaitility.Awaitility.await;
-import static com.jayway.restassured.RestAssured.get;
 import static com.jayway.restassured.RestAssured.given;
 import static com.jayway.restassured.RestAssured.when;
 import static org.hamcrest.Matchers.containsString;
 
 /**
  * @author Heiko Braun
  */
+@RunWith(Arquillian.class)
 public class OpenshiftIT {
 
-    private static final OpenShiftTestAssistant openshift = new OpenShiftTestAssistant();
+    @RouteURL("wfswarm-rest-http")
+    private URL url;
 
-    @BeforeClass
-    public static void setup() throws Exception {
-        // the application itself
-        openshift.deployApplication();
-
-        // wait until the pods & routes become available
-        openshift.awaitApplicationReadinessOrFail();
-
-        await().atMost(5, TimeUnit.MINUTES).until(() -> {
```

Comment:
My experience has shown that it's better to test the response and not wait only for the "readiness". Something things are wild for a few seconds.

Reply:
(this is why we have the 2 checks BTW)

Reply:
Cube just awaits readiness, true. Could add that back, but I didn't see a failure yet. Obviously, everything works fine locally...